### PR TITLE
export all peripherals

### PR DESCRIPTION
**Pull request by hackaugusto**
My motivation for this was to use the SPI. I'm not sure if the idea was not to export all peripherals just to keep the API clean (or for some other reason), so fell free to close the PR if this doesn't fit the crate's goals.

### CODE REVIEW

**Review by BartMassey — COMMENTED** [review]
I think some of these peripherals are "legacy" and are included for back-compatibility: better alternatives are available. Examples are UART (vs UARTE), TWI (vs TWIM), SPI (vs SPIM). While we should still definitely import these, it would be nice to mark this somehow (probably in the doc comment) to avoid very minor footguns.

**Comment by qwandor** [comment]
Are all of these peripherals actually usable, on the microbit, i.e. connected to something electrically?

File: microbit-common/src/v2/board.rs
[the remark this answers is not in the document]
Comment (hackaugusto) [reply microbit-common/src/v2/board.rs:1]: Added the the peripherals to v1 too 👍

**Comment by hackaugusto** [comment]
> I think some of these peripherals are "legacy" and are included for back-compatibility: better alternatives are available. Examples are UART (vs UARTE), TWI (vs TWIM), SPI (vs SPIM). While we should still definitely import these, it would be nice to mark this somehow (probably in the doc comment) to avoid very minor footguns.

I'm not sure which peripherals are considered legacy, should I only add comments for the ones you mentioned?

**Review by qwandor — APPROVED** [review]

**Pull request merged**